File: 프로그래머스/2/152996. 시소 짝꿍/시소 짝꿍.cs
using System;
using System.Collections.Generic;

public class Solution {
    public long solution(int[] weights) {
        long answer = 0;
        
        Array.Sort(weights);
        
        var db = new Dictionary<double, long>();
        
        foreach(var w in weights)
        {
            if(db.ContainsKey(w)) answer += db[w];
            
            if((w * 2) % 3 == 0)
            {
                int target = (w * 2) / 3;
                if(db.ContainsKey(target)) answer += db[target];
            }
            
            if(w % 2 == 0)
            {
                int target = w / 2;
                if(db.ContainsKey(target)) answer += db[target];
            }
            
            if((w * 3) % 4 == 0)
            {
                int target = (w * 3) / 4;
                if(db.ContainsKey(target)) answer += db[target];
            }
            
            if(db.ContainsKey(w)) db[w]++;
            else db[w] = 1;
        }
        return answer;
    }
}

File: 프로그래머스/2/152996. 시소 짝꿍/시소 짝꿍.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

long long solution(vector<int> weights) {
    long long answer = 0;
    
    sort(weights.begin(), weights.end());
    
    unordered_map<double, long long> mp;
    
    for(int w : weights)
    {
        answer += mp[w];
        
        answer += mp[w / 2.0];
        
        answer += mp[w * 2.0 / 3.0];
        
        answer += mp[w * 3.0 / 4.0];
        
        mp[w]++;
    }
    return answer;
}